firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -318,6 +314,19 @@ private InstallationTokenResult readGenerateAuthTokenResponse(HttpsURLConnection
     return builder.build();
   }
 
+  // Read the error message from the response.
+  private String readErrorResponse(HttpsURLConnection conn) throws IOException {
+    BufferedReader reader =
+        new BufferedReader(new InputStreamReader(conn.getErrorStream(), Charset.defaultCharset()));
```

Comment:
shouldn't we use UTF8 here instead? are we expecting the backend to return a default-charset encoded response or a utf8 one?